### PR TITLE
[rllib] Update concepts docs and add "Building Policies in Torch/TensorFlow" section

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This documents changes from https://github.com/ray-project/ray/pull/4795 and https://github.com/ray-project/ray/pull/4819/files and https://github.com/ray-project/ray/pull/4820/files.

It can probably be merged independently of those PRs.

### CODE REVIEW

File: doc/source/rllib-concepts.rst [hunk 1]
```diff
@@ -45,37 +46,356 @@ Most interaction with deep learning frameworks is isolated to the `Policy interf
         def set_weights(self, weights):
             self.w = weights["w"]
 
+Building Policies in TensorFlow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (richardliaw) [inline doc/source/rllib-concepts.rst]: you can do `----------------` for another level of nesting below

File: doc/source/rllib-concepts.rst [hunk 2]
```diff
@@ -45,37 +46,356 @@ Most interaction with deep learning frameworks is isolated to the `Policy interf
...
+
+This section covers how to build a TensorFlow RLlib policy using ``tf_policy_template.build_tf_policy()``.
```
Comment (richardliaw) [inline doc/source/rllib-concepts.rst]: I would actually make this a top level page: `RLlib Example: Building a Policy`. This is pretty hard to find given that it's under `RLlib Concepts and Custom Algorithms`, which isn't something that is immediately reminiscent.

File: doc/source/rllib-concepts.rst [hunk 3]
```diff
@@ -45,37 +46,356 @@ Most interaction with deep learning frameworks is isolated to the `Policy interf
...
+    MyTFPolicy = build_tf_policy(
+        name="MyTFPolicy",
+        loss_fn=policy_gradient_loss,
+        postprocess_fn=postprocess_advantages)
+
+The ``postprocess_advantages()`` function above uses calls RLlib's ``compute_advantages`` function to compute advantages for each timestep. If you re-run the trainer with this improved policy, you'll find that it quickly achieves the max reward of 200.
+
+You might be wondering how RLlib makes the advantages placeholder automatically available as ``batch_tensors[Postprocessing.ADVANTAGES]``. When building your policy, RLlib will create a "dummy" trajectory batch where all observations, actions, rewards, etc. are zeros. It then calls your ``postprocess_fn``, and generates TF placeholders based on the numpy shapes of the postprocessed batch. RLlib tracks which placeholders that ``loss_fn`` and ``stats_fn`` access, and then feeds the corresponding sample data into those placeholders during loss optimization. You can also access these placeholders via ``policy.get_placeholder(<name>)`` after loss initialization.
+
+**Example 1: Proximal Policy Optimization**
```
Comment (richardliaw) [inline doc/source/rllib-concepts.rst]: (Or make this a top-level thing, under `Examples`)

File: doc/source/rllib-concepts.rst (hunk 2 above)
Comment (ericl) [reply doc/source/rllib-concepts.rst]: I renamed the page title. We can revisit after DQN is also fleshed out, if it makes sense to have a separate page from concepts (not sure if we need a concepts page?)

**Review by richardliaw — APPROVED** [review]

**Pull request merged**